packages/icon-build-helpers/src/builders/react/next.js

```diff
@@ -0,0 +1,256 @@
+/**
+ * Copyright IBM Corp. 2019, 2019
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+const t = require('@babel/types');
+const { default: generate } = require('@babel/generator');
+const { default: template } = require('@babel/template');
+const { babel } = require('@rollup/plugin-babel');
+const fs = require('fs-extra');
+const path = require('path');
+const { rollup } = require('rollup');
+const virtual = require('../plugins/virtual');
+const { babelConfig } = require('./next/babel');
+const { svgToJSX, jsToAST } = require('./next/convert');
+const templates = require('./next/templates');
+
+// This builder outputs a collection of CommonJS modules representing our icon
+// components. It does not generate an `index.js` entrypoint file due to the
+// number of icons we currently support (>1600)
+//
+// Each CommonJS module includes an icon component. For more information on the
+// structure of each module, checkout `createIconEntrypoint`
+async function builder(metadata, { output }) {
+  const modules = metadata.icons.map((icon) => {
+    const { moduleInfo } = icon;
+    return {
+      filepath: moduleInfo.filepath,
+      moduleName: moduleInfo.name,
+      entrypoint: createIconEntrypoint(
+        moduleInfo.local,
+        moduleInfo.sizes,
+        icon.deprecated,
+        icon.reason
+      ),
+    };
+  });
+
+  // Rollup allows us to define multiple "entrypoints" instead of only one when
+  // creating a bundle. This allows us to map different input paths in the
+  // `input` object to files that we're generating for each icon component in
+  // the `files` object
+  const files = {};
+  const input = {};
+
+  for (const m of modules) {
+    files[m.filepath] = m.entrypoint;
+    input[m.filepath] = m.filepath;
+  }
+
+  const bundle = await rollup({
+    input,
+    external: ['@carbon/icon-helpers', 'react', 'prop-types'],
+    plugins: [
+      // We use a "virtual" plugin to pass all of our components that we
+      // created from our metadata to rollup instead of rollup trying to read
+      // these files from disk
+      virtual({
+        // Each of our Icon modules use the "./Icon.js" path to import this base
+        // componnet
+        './Icon.js': await fs.readFile(
+          path.resolve(__dirname, './components/Icon.js'),
+          'utf8'
+        ),
+        './iconPropTypes.js': `
+          import PropTypes from 'prop-types';
+
+          export const iconPropTypes = {
+            size: PropTypes.oneOfType([
+              PropTypes.number,
+              PropTypes.string,
```

Comment:
`size` accepts strings but `width` and `height` seem to take numbers only

![image](https://user-images.githubusercontent.com/8265238/113313275-b60b9f80-92d0-11eb-8d97-9c116b133838.png)

Reply:
Thanks @emyarod! Great catch. Updated the `Icon.js` prop type to include number/string which should handle this.